app/app_xdg.go

```diff
@@ -122,5 +228,49 @@ func rootCacheDir() string {
 }
 
 func watchTheme() {
-	// no-op, not able to read linux theme in a standard way
+	wm := findCurrentWM()
+	switch wm {
+	case "gnome":
+		go watchGnomeTheme()
+	case "kde":
+		// no-op, not able to read linux theme in a standard way
+	}
+}
+
+func watchGnomeTheme() {
+	// connect to dbus to detect color-schem theme changes
+	conn, err := dbus.SessionBus()
+	if err != nil {
+		log.Println(err)
+		return
+	}
+
+	if err := conn.AddMatchSignal(
+		dbus.WithMatchObjectPath("/org/freedesktop/portal/desktop"),
+		dbus.WithMatchInterface("org.freedesktop.portal.Settings"),
+		dbus.WithMatchMember("SettingChanged"),
+	); err != nil {
+		log.Println(err)
+		return
+	}
+	defer conn.Close()
+	dbusChan := make(chan *dbus.Signal, 10)
+	conn.Signal(dbusChan)
+	currentTheme := fyne.CurrentApp().Settings().Theme()
+	for sig := range dbusChan {
+		for _, v := range sig.Body {
+			switch v {
+			case "color-scheme":
+				if currentTheme == theme.DefaultTheme() {
+					variant := findGnomeThemeVariant()
+					switch variant {
+					case theme.VariantLight:
+						fyne.CurrentApp().Settings().SetTheme(theme.LightTheme())
```

Comment:
This (and the line 2 down) is wrong - it will override developer set themes.
Try to copy the darwin code (`app_desktop_darwin.go`) and use the call to `setupTheme()` instead.

Reply:
I made variation, the darwin code calls C implementaition.

Reply:
Ah yes indeed, good implementation for pure Go version